Dockerfile

```
@@ -0,0 +1,30 @@
+FROM golang:1.20-alpine
+
+RUN apk add --no-cache \
+  git \
+  gcc \
+  g++ \
+  musl-dev
+
+#ENV CC=x86_64-linux-musl-gcc
+#ENV CXX=x86_64-linux-musl-g++
+ARG MACHINE_ID="nothing"
+
+ENV GOARCH=amd64
+ENV GOOS="linux"
+ENV CGO_ENABLED=1
+
+WORKDIR /src
+
+RUN git clone --depth=1 https://github.com/maxpert/marmot
+WORKDIR /src/marmot
```

Ideally you wouldn't clone the repo here but just `COPY . /src` so that Docker's layer caching can actually work.

```suggestion
COPY . /src

WORKDIR /src/marmot
```